lighthouse-core/gather/computed/trace-of-tab.js

```diff
@@ -94,8 +93,7 @@ class TraceOfTab extends ComputedArtifact {
     // subset all trace events to just our tab's process (incl threads other than main)
     // stable-sort events to keep them correctly nested.
     const processEvents = trace.traceEvents
-      .filter(e => e.pid === startedInPageEvt.pid)
-      .stableSort((event0, event1) => event0.ts - event1.ts);
+      .filter(e => e.pid === startedInPageEvt.pid);
```

Comment:
I guess my issues here are:
- trace events aren't necessarily in order
- our finding of main thread, the correct navStart etc aren't perfect, but I'm concerned that all of our logic for those (from `keyEvents` above) currently relies on the order after sorting :) 
- I don't believe any gatherer/audit *currently* relies on correctly (stably) sorted events -- everything looking at the trace is looking at top level events, but it feels like something we're just going to have to re-add in the future when we start looking at nested events
- no sort at all seems really wrong, though. TTFI and TTCI currently rely on the top level events being in order (from `getMainThreadTopLevelEvents` which gets its order from `mainThreadEvents`)...that may be true most of the time but isn't necessarily, and with the natural noise in those metrics it would be difficult to spot the bug if it does cause issues

since the absolute time spent on this is pretty small (like 900ms on a big trace I just tried vs like 500ms with native sort), it seems like the main benefit here is for the mass processing of traces rather than just a Lighthouse run, so personally I feel like we should keep the stable sort.

So we could:
- keep `stableSort`
- `stableSort` only `keyEvents` (should be fast because there aren't that many key events) and warn everywhere `processEvents` and `mainThreadEvents` aren't sorted (and TTFI and TTCI could add their own sorts after extracting quiet periods or whatever)
- Switch to timsort. We discussed using timsort a while back because it's stable but is also optimized for mostly or partly sorted data, which fits the trace events use case well. Using https://www.npmjs.com/package/timsort I sorted the same trace in about 300ms, which seems like it could be a good compromise with "no sort at all" :) One problem is [one longstanding issue](https://github.com/mziccard/node-timsort/issues/14) open against the repo, but at worst we could wrap the sort line with a try/catch and fall back to devtool's `stableSort` if timsort throws (and maybe log to sentry to see if it's ever actually a problem for us...as @paulirish notes, we have the simplest case possible since we're only sorting integer timestamps)

Reply:
I can revert for now and we can punt for later. Maybe we end up fixing https://github.com/mziccard/node-timsort/issues/14 and use it?

Reply:
> I can revert for now and we can punt for later.

sgtm

> Maybe we end up fixing mziccard/node-timsort#14 and use it?

I like that plan :)